components/datadog/agentparams/filepermissions/permissions.go

```diff
@@ -0,0 +1,8 @@
+package filepermissions
+
+// FilePermissions interface defines all commands that can be used to set and reset file permissions.
+// These functions are used to create args.Command that will be run by pulumi.
+type FilePermissions interface {
+	SetupPermissionsCommand(path string) string
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm thinking it could be eg. `SetupPermissions(path string, executor interface{Execute() error})` so that we can do some more complex conditions on the commands (parse command output, do different things depending on errors, etc)

But no issue if you want to keep the current version, that works too

Reply:
Hmm sounds like an interesting idea, but I'll rather keep this PR simple and add `executor` later when we have a use case for it